test/integration/RedisLocker.test.ts

```diff
@@ -201,19 +201,20 @@ describeIf('docker', 'A server with a RedisLocker', (): void => {
           return num;
         }));
 
-      await flushPromises();
+      setTimeout((): void => {
+        emitter.emit('release1');
+        emitter.emit('release0');
+        emitter.emit('release2');
+      }, 200);
```

Comment:
Since this is an integration test that has no timer mocking, this will always cause this part to take 200ms and makes our test dependent on the timer.

I'm thinking that perhaps instead we could have a solution where `withReadLock` triggers something when entered so we can force the order like that such as firing an event that is being listened to or calling a function.